### PR TITLE
Update IAST unvalidated redirect vulnerability

### PR DESCRIPTION
**Pull request by jandro996**
## Description

- Update sink test for unvalidated redirect vulnerability
- Add jersey and resteasy endpoints to test unvalidated redirect in header and location Response methods
- Add servlet forward test

## Motivation

Test unvalidated redirect vulnerability

### CODE REVIEW

File: tests/appsec/iast/sink/test_unvalidated_redirect.py [hunk 1]
```diff
@@ -21,15 +43,23 @@ class TestUnvalidatedRedirect:
...
     sink_fixture_redirect = SinkFixture(
         vulnerability_type="UNVALIDATED_REDIRECT",
         http_method="POST",
         insecure_endpoint="/iast/unvalidated_redirect/test_insecure_redirect",
         secure_endpoint="/iast/unvalidated_redirect/test_secure_redirect",
         data={"location": "http://dummy.location.com"},
-        location_map={"java": {"spring-boot": "com.datadoghq.system_tests.springboot.AppSecIast"}},
+        location_map=_expected_location,
+    )
+    sink_fixture_forward = SinkFixture(
```
Comment (smola) [inline tests/appsec/iast/sink/test_unvalidated_redirect.py:56]: We could just have a different file and class for `test_unvalidated_forward.py`, I'm fairly sure that the `@release` decorator and implementation timeframes will generally be quite different for both across languages.

**Review by smola — CHANGES_REQUESTED** [review]

**Review by smola — APPROVED** [review]

**Review by cbeauchesne — COMMENTED** [review]
I see some errors on CI, is it related ?

**Comment by jandro996** [comment]
> I see some errors on CI, is it related ?

I think that it's not related but let me update the PR and check again 👍

**Comment by jandro996** [comment]
@cbeauchesne I'm pretty sure that these fails are not related with the changes in this PR. It seems that many system-test PRs are failing due to the failing tasks right now

**Review by cbeauchesne — APPROVED** [review]

File: tests/appsec/iast/sink/test_unvalidated_redirect_forward.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "spring-boot-jetty": "?",
+        "spring-boot-openliberty": "?",
+        "spring-boot-wildfly": "?",
+        "spring-boot-undertow": "?",
+        "resteasy-netty3": "?",
+        "jersey-grizzly2": "?",
+        "*": "?",
+    }
+)
+class TestUnvalidatedRedirect:
```
Comment (smola) [inline tests/appsec/iast/sink/test_unvalidated_redirect_forward.py:37]:
```suggestion
class TestUnvalidatedForward:
```

**Review by smola — CHANGES_REQUESTED** [review]

**Review by smola — APPROVED** [review]

**Pull request merged**